### PR TITLE
TUNIC: Update victory condition

### PR DESCRIPTION
**Pull request by silent-destroyer**
## What is this fixing or adding?
This adds the hero relics to the victory condition when needing to defeat the final boss.
They aren't added if hexagon quest is on since you don't fight the final boss for that goal.
Also changes the ability unlock pages to not be marked as progression if ability shuffle is turned off.

## How was this tested?
Running test gens and checking spoiler logs

### CODE REVIEW

File: worlds/tunic/er_rules.py [hunk 1]
```diff
@@ -991,7 +992,8 @@ def set_er_region_rules(world: "TunicWorld", ability_unlocks: Dict[str, int], re
         connecting_region=regions["Spirit Arena Victory"],
         rule=lambda state: (state.has(gold_hexagon, player, world.options.hexagon_goal.value) if
                             world.options.hexagon_quest else
-                            state.has_all({red_hexagon, green_hexagon, blue_hexagon, "Unseal the Heir"}, player)))
+                            (state.has_all({red_hexagon, green_hexagon, blue_hexagon, "Unseal the Heir"}, player)
+                            and state.has_all(hero_relics, player))))
```
Comment (ScipioWright) [inline worlds/tunic/er_rules.py:996]:
```suggestion
                            and state.has_group_unique("Hero Relics", player, 6))))
```

File: worlds/tunic/rules.py [hunk 2]
```diff
@@ -128,7 +129,7 @@ def set_region_rules(world: "TunicWorld", ability_unlocks: Dict[str, int]) -> No
         or has_ice_grapple_logic(False, state, player, options, ability_unlocks)
     multiworld.get_entrance("Overworld -> Spirit Arena", player).access_rule = \
         lambda state: (state.has(gold_hexagon, player, options.hexagon_goal.value) if options.hexagon_quest.value
-                       else state.has_all({red_hexagon, green_hexagon, blue_hexagon}, player)) and \
+                       else state.has_all({red_hexagon, green_hexagon, blue_hexagon}, player) and state.has_all(hero_relics, player)) and \
```
Comment (ScipioWright) [inline worlds/tunic/rules.py:132]:
```suggestion
                       else state.has_all({red_hexagon, green_hexagon, blue_hexagon}, player) and state.has_group("Hero Relics", player, 6)) and \
```

**Review by ScipioWright — APPROVED** [review]
LGTM, gens fine

**Review by Exempt-Medic — APPROVED** [review]
The changes LGTM and make sense with my understanding of the game. Merged into `main` and it generates just fine and saw that the items were being required and created appropriately.

Only thing that I think is a tiny bit strange is using `> 0` for abilities since if it's greater than 1, that might be weird for filler creation, but I also can't think of a reason these would be greater than 1 in the first place, so it seems fine.

**Comment by silent-destroyer** [comment]
> The changes LGTM and make sense with my understanding of the game. Merged into `main` and it generates just fine and saw that the items were being required and created appropriately.
> 
> Only thing that I think is a tiny bit strange is using `> 0` for abilities since if it's greater than 1, that might be weird for filler creation, but I also can't think of a reason these would be greater than 1 in the first place, so it seems fine.

Yeah, '== 1' might've been better here but there should only ever be able to be one of each page in the item pool anyway. That check is to mainly ensure that they're still supposed to be created at all (since in the case of hexagon hunt they get removed from the pool entirely). The pages aren't considered filler so there shouldn't be any chance of them being created multiple times

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**